src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DataAnnotationsClientModelValidatorProvider.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    /// <summary>
+    /// An implementation of <see cref="IClientModelValidatorProvider"/> which providers client validators
+    /// for attributes which derive from <see cref="ValidationAttribute"/>. It also provides
+    /// a validator for types which implement <see cref="IClientModelValidator"/>.
+    /// The logic to support <see cref="IClientModelValidator"/>
+    /// is implemented in <see cref="DataAnnotationsClientModelValidator"/>.
+    /// </summary>
+    public class DataAnnotationsClientModelValidatorProvider : IClientModelValidatorProvider
```

Comment:
In terms of layering - can all of the client validation move to Rendering? It doesn't have much to do with model binding or the validation system in there now, does it?